### ISSUE TITLE
[subplat] Email Communication - Firefox News option is not checked if user accepted to receive newsletter

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#10168]
**Affected browser versions:**

* All Chrome;

**Affected FxA versions:**

* 1.213.1;

**Prerequisites:**

* Have a VPN set to Germany location;
* Have a subscription made for a passwordless account and check to receive newsletter;

**Steps to reproduce:**
1. Go to your email and click the “Create password” button;
2. Finalize account creation;
3. Load Firefox Account page using the account from prerequisites;
4. Select “Email Communication”;
5. Check “Firefox News" option;

**Expected result:**

* The "Firefox News" option is checked.

**Actual result:**

* The "Firefox News" option is not checked.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-3865)

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Thanks Valentina for filing this issue.

TL;DR: QA cannot verify this test case currently in FxA stage. Consequently, no QA verification is needed for this ticket; I will verify the fix in stage and prod. A more detailed explanation follows…

It turns out there were three separate issues at play here:

  1. FxA’s “Email Communications” link in FxA Settings pulls information from a different server (Basket stage) than where we are actually recording the newsletter signup (Basket dev).
  2. Newsletter information from checkout was being sent in the wrong format, and
  3. We weren’t validating the format of the newsletter information passed into the back-end, so it was incorrectly responding with a 200.

I fixed 2) and 3), but unfortunately 1) means that it is not currently possible for QA to verify this fix in Stage. In fact, this is true even for existing newsletter signup entry points like when creating an FxA account. 

Consequently, for this epic, I will own verifying this fix in stage and prod, and there is no further action needed by QA.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Verified on stage 1.213.3 with the curl command and dev API key from the Marketing team.

When I look up a test user on basket-dev, the list of newsletters correctly shows the expected value: ["mozilla-and-you"].
